nixos/modules/programs/java.nix

```diff
@@ -40,12 +38,38 @@ in
         type = types.package;
       };
 
+      binfmt = mkOption {
+        type = types.bool;
+        default = false;
+        description = lib.mdDoc "Register Java program execution with binfmt";
+      };
+
     };
 
   };
 
   config = mkIf cfg.enable {
 
+    boot.binfmt.registrations = mkIf cfg.binfmt {
+      java-class = {
+        recognitionType = "extension";
+        magicOrExtension = "class";
+        interpreter = pkgs.writeShellScript "java-class-wrapper" ''
+          file=''${1%%.class}
+          file=''${file/.\//}
```

Comment:
Is this just to remove a `./` at the beginning? Shouldn't this use `basename "$1"` instead and add `dirname "$1"` to `CLASSPATH`?

Reply:
It removes `.class` from the filename.

Reply:
The first `file=` line does that right? I'm talking about the second

Reply:
Not sure.

Reply:
`basename` just retrieves the filename in a path, `dirname` does that for the path.

Reply:
Alright, newest commit has that now.